include/eve/detail/function/simd/x86/load.hpp

```diff
@@ -72,4 +86,36 @@ namespace eve::detail
       }
     }
   }
+
+  //================================================================================================
+  // logical loads require some specific setup
+  //================================================================================================
+  template<typename T, typename N, typename Ptr, x86_abi ABI>
+  EVE_FORCEINLINE
+  auto load(eve::as_<logical<wide<T, N, ABI>>> const &, ABI const & abi, Ptr p)
+  requires( std::same_as<logical<T>, std::remove_cvref_t<decltype(*p)>> )
+  {
+    auto block = [&]() -> wide<T, N, ABI>
+    {
+      using tgt = eve::as_<wide<T, N, ABI>>;
+      if constexpr( !std::is_pointer_v<Ptr> ) return load(tgt(), abi, (T const*)(p.get()));
+      else                                    return load(tgt(), abi, (T const*)(p));
```

Comment:
Why do you need to do the dispatch here? Won't the `load` you call figure out the dispatch?

Reply:
This load is the internal one called by wide constructor. The overload system is not operating at this point yet.
This is one thing we need to live with.